micro-infra-spring-config/src/main/java/com/ofg/infrastructure/property/AppCoordinates.java

```diff
@@ -39,8 +41,29 @@ public static AppCoordinates defaults(Resource microserviceConfigResource) {
         this.countryCode = requireNonNull(countryCode);
     }
 
-    public File getConfigFolder(File rootFolder) {
+    public ConfigLocations getConfigLocations(File rootFolder) {
+        return new ConfigLocations(
+                getCommonConfigFolder(rootFolder),
+                getEnvConfigFolder(rootFolder),
+                getCountryConfigFolder(rootFolder)
+        );
+    }
+
+    private File getCommonConfigFolder(File rootFolder) {
+        File folder = new File(rootFolder, COMMON_DIR);
+        return addNameComponentsToFolder(folder);
+    }
+
+    private File getEnvConfigFolder(File rootFolder) {
         File folder = new File(rootFolder, environment);
+        return addNameComponentsToFolder(folder);
+    }
+
+    private File getCountryConfigFolder(File rootFolder) {
+        return new File(getEnvConfigFolder(rootFolder), countryCode);
+    }
+
+    private File addNameComponentsToFolder(File folder) {
         final String[] components = nameComponents();
         for (int i = 0; i < components.length - 1; i++) {
```

Comment:
What was a reason to skip the last component? For people reading that code it would not be probably very clear.